src/main/java/de/internetsicherheit/brl/bloxberg/cache/ethereum/BloxbergClient.java

```diff
@@ -52,15 +53,43 @@ private EthBlock getEthBlock(BigInteger block) throws IOException {
         return web3j.ethGetBlockByNumber(DefaultBlockParameter.valueOf(block), true).send();
     }
 
-    public BlockWithData getBlockWithData(BigInteger block) throws IOException {
-        this.rawBlock = this.getEthBlock(block);
-        this.blockWithData = new BlockWithData(this.rawBlock.getBlock());
+    private Block getBlockWithData(BigInteger block) throws IOException {
+        EthBlock rawBlock = this.getEthBlock(block);
+        this.blockWithData = new Block(rawBlock.getBlock());
         return this.blockWithData;
     }
 
-    public BigInteger getBlockTimestamp(BigInteger block) throws IOException {
-        BlockWithData blockForTimestamp = this.getBlockWithData(block);
-        return blockForTimestamp.getTimestamp();
+    private boolean isSmartContract(String address) throws IOException {
+
+        EthGetCode addressCodeRequest = web3j.ethGetCode(address, DefaultBlockParameter.valueOf("latest")).send();
+        return !addressCodeRequest.getResult().equals("0x");
+    }
+
+
+    private TransactionAddress.type getTransactionAddressType(TransactionAddress address) throws IOException {
+        if (address.address == null) {
+            return TransactionAddress.type.NONE;
+        } else if (isSmartContract(address.address)) {
+            return TransactionAddress.type.SMARTCONTRACT;
+        } else {
+            return TransactionAddress.type.USERORVALIDATOR;
+        }
+    }
+
+
+    private void labelTransactionsOfBlock(Block block) throws IOException {
```

Comment:
Something like this as a `void` method feels like a smell, this method is just performing a side effect on the state of the provided method argument, this is not a good style.

Reply:
I will give a try to doing the changes by myself,and asking in the corresponding change if something is unclear to me :)